test_player/views.py

```diff
@@ -0,0 +1,352 @@
+         answers related with current question;
+
+        :var current_question(object): single object of question by lookup
+         parameters from form keyword arguments such as quiz_id,
+         question_number;
+        :var current_answers(queryset): queryset object with answers related
+         with current question;
+        :return: kwargs(dict): with updated form instance keyword arguments.
+
+        """
+        kwargs = super(TestPlayer, self).get_form_kwargs()
+        current_quiz, question_number = self.get_quiz_data()
+        current_question = get_object_or_404(Questions,
+                                             quiz_id=current_quiz,
+                                             question_number=question_number)
+        current_answers = current_question.answers_set.all()
+
+        kwargs.update({'answers': current_answers})
+
+        return kwargs
+
+    def form_valid(self, form):
+        """
+        :param form
+        :var answer(int): selected answer for current question
+        :return form_valid(form).
+
+        """
+        current_quiz, question_number = self.get_quiz_data()
+        answer = form.cleaned_data.get('answers')
+        # checking if session variable exist, if not - call method, which \
+        # set default initial values (used only once during the first request)
+        if current_quiz not in self.request.session:
+            self._setting_initial_session_data()
+
+        # updating session with relevant data
+        session_data = self.request.session[current_quiz]
+        if form.has_changed():
+            session_data.update(
+                {question_number: answer}
+            )
+        self.request.session[current_quiz] = session_data
+
+        return super(TestPlayer, self).form_valid(form)
+
+    def _setting_initial_session_data(self):
+        """
+        Method, which set default initial values (used only once during the
+         first request).
+
+        :var questions(queryset object): which contain ordered questions
+         numbers;
+        :var initial_questions(dict): which contain initial key(questions),
+         values(answers) for questions.
+
+        """
+        current_quiz, _ = self.get_quiz_data()
+
+        questions = Questions.objects.filter(quiz=current_quiz).values_list(
+            'question_number', flat=True).order_by('question_number')
+        initial_questions = dict.fromkeys(questions)
+
+        self.request.session.setdefault(current_quiz, None)
+        self.request.session[current_quiz] = initial_questions
+
+    def _saving_user_answers(self, current_quiz):
+        """
+        Method, which save user answers to database such as json object.
+
+        :var session_data(dict): with session data from session object;
+        :var session_user(int): current logged in user who passes test.
+
+        """
+        session_user = self.request.session['_auth_user_id']
+        session_data = dict(self.request.session[current_quiz])
```

Comment:
Are you sure you need to use dict method?

Reply:
done.